core/domain/topic_jobs_one_off.py

```diff
@@ -92,3 +94,68 @@ def reduce(key, values):
                 sum(ast.literal_eval(v) for v in values))])
         else:
             yield (key, values)
+
+
+class RemoveDeletedUncategorizedSkillsOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """One-off job to remove deleted uncategorized skills linked to a topic."""
+
+    _DELETED_KEY = 'topic_deleted'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [topic_models.TopicModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            yield (RemoveDeletedUncategorizedSkillsOneOffJob._DELETED_KEY, 1)
+            return
+
+        # Note: the read will bring the topic up to the newest version.
+        topic = topic_fetchers.get_topic_by_id(item.id)
+        skill_ids_to_be_removed_from_subtopic = []
+        all_skill_ids_to_be_removed = []
+        commit_cmds = []
+        for subtopic in topic.get_all_subtopics():
+            local_skill_models = skill_models.SkillModel.get_multi(
+                subtopic['skill_ids'])
+            for skill_id, skill_model in python_utils.ZIP(
+                    subtopic['skill_ids'], local_skill_models):
+                if skill_model is None:
+                    commit_cmds.append(topic_domain.TopicChange({
+                        'cmd': topic_domain.CMD_REMOVE_SKILL_ID_FROM_SUBTOPIC,
+                        'skill_id': skill_id,
+                        'subtopic_id': subtopic['id']
+                    }))
+                    skill_ids_to_be_removed_from_subtopic.append(skill_id)
+
+        local_skill_models = skill_models.SkillModel.get_multi(
+            topic.get_all_skill_ids())
+        list_of_all_skill_ids = (
```

Comment:
This doesn't seem like "all skill ids" to me... it omits the skill ids assigned to subtopics, right? (The naming might be incorrect.)

Reply:
It contains skill ids that were unassigned from subtopics. Please see this [comment](https://github.com/oppia/oppia/pull/9291#discussion_r422966124) for a detailed explanation of what is happening here.